dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -231,27 +235,51 @@ public static HashSet<string> InferInputs(Template template)
             return visitor._variables;
         }
 
-        public override void Visit(ScriptVariableGlobal node)
+        public override Statement Visit(Statement statement)
```

Comment:
Nice! This could almost be part of a default visitor, but at the same time it's easy to implement. What I was suggesting wat to override `VisitForExpression` (and also VisitAssignExpression) but this technique is more generic and works as well.